### PR TITLE
security: updating semver to 7.5.4 to resolve CVE-2022-25883

### PR DESCRIPTION
**Pull request by jakelacey2012**
By submitting a PR to this repository, you agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md). Please see the [contributing guidelines](https://github.com/auth0/.github/blob/master/CONTRIBUTING.md) for how to create and submit a high-quality PR for this repo.

### Description

This PR updates [semver](https://www.npmjs.com/package/semver) to a minimum version of 7.5.4 to resolve CVE-2022-25883.

### References

- CVE-2022-25883
- https://github.com/auth0/node-jsonwebtoken/issues/921
- https://github.com/auth0/node-jsonwebtoken/pull/919

### Testing

- ✅  `npm test` is passing.
- ✅  This change adds test coverage for new/changed/fixed functionality

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR or in auth0.com/docs
- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used, if not the default branch

### CODE REVIEW

**Review by david-renaud-okta — APPROVED** [review]

**Comment by SEKERM** [comment]
please go to recent version 7.5.4.

**Review by SEKERM — COMMENTED** [review]
please go to 7.5.4.

**Comment by mkrudele** [comment]
When will this PR be merged and a new version of the module published?

File: package.json [hunk 1]
```diff
@@ -39,7 +39,7 @@
     "jws": "^3.2.2",
     "lodash": "^4.17.21",
     "ms": "^2.1.1",
-    "semver": "^7.3.8"
+    "semver": "^7.5.2"
```
Comment (cpettet) [inline package.json:42]:
```suggestion
    "semver": "^7.5.4"
```

Comment (cpettet) [reply package.json:42]: There's a few bugfixes from [7.5.3](https://github.com/npm/node-semver/releases/tag/v7.5.3) and [7.5.4](https://github.com/npm/node-semver/releases/tag/v7.5.4).

**Comment by jakelacey2012** [comment]
@SEKERM @cpettet thanks for the feedback, I've updated to 7.5.4.

**Review by SEKERM — APPROVED** [review]
lgtm. @david-renaud-okta  can you give approval with write access:

Review required
At least 1 approving review is required by reviewers with write access.

**Review by CharlesRea — APPROVED** [review]

**Comment by SEKERM** [comment]
@jakelacey2012 when will you merge into main? Thank you.

**Comment by Uzlopak** [comment]
You could have just removed semver

https://github.com/auth0/node-jsonwebtoken/pull/880/files 

D'oh

**Pull request merged**